src/sections/user/view/user-view.tsx

```diff
@@ -24,15 +26,144 @@ import { emptyRows, applyFilter, getComparator } from '../utils';
 
 import type { UserProps } from '../user-table-row';
 
+import { UserAddModal } from '../user-add-modal';
+import { UserEditModal } from '../user-edit-modal';
+import { ConfirmDeleteDialog } from '../confirm-delete-dialog';
 // ----------------------------------------------------------------------
 
 export function UserView() {
   const table = useTable();
 
   const [filterName, setFilterName] = useState('');
 
+  const [users, setUsers] = useState<UserProps[]>(_users); // State to store users
+  const [isAddModalOpen, setAddModalOpen] = useState(false); // State to control modal
+  
+  const [isEditModalOpen, setEditModalOpen] = useState(false); // State to control Edit Modal
+  const [currentUser, setCurrentUser] = useState<UserProps | null>(null); // Store current user to edit
+
+  const [isDeleteDialogOpen, setDeleteDialogOpen] = useState(false); // State for delete dialog
+  const [userToDelete, setUserToDelete] = useState<UserProps | null>(null); // Track the user to delete
+
+  // Snackbar states
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState('');
+  const [snackbarSeverity, setSnackbarSeverity] = useState<'success' | 'error'>('success');
+
+  // Snackbar handlers
+  const openSnackbar = (message: string, severity: 'success' | 'error') => {
+    setSnackbarMessage(message);
+    setSnackbarSeverity(severity);
+    setSnackbarOpen(true);
+  };
+
+  const handleSnackbarClose = () => {
+    setSnackbarOpen(false);
+  };
+
+
+  const handleOpenAddModal = () => {
+    console.log('Opening Modal');
+    setAddModalOpen(true);
+  };
+  const handleCloseAddModal = () => setAddModalOpen(false);
+
+  const handleAddUser = (newUser: UserProps) => {
+    setUsers((prevUsers) => [...prevUsers, newUser]);
+  };
+
+  const handleOpenEditModal = (user: UserProps) => {
+    setCurrentUser(user); // Set the user to be edited
+    setEditModalOpen(true);
+  };
+  const handleCloseEditModal = () => setEditModalOpen(false);
+
+  const handleEditUser = (updatedUser: UserProps) => {
+    setUsers((prevUsers) =>
+      prevUsers.map((user) =>
+        user.id === updatedUser.id ? updatedUser : user
+      )
+    );
+  };
+
+  // const handleDeleteUser = (userId: string) => {
+  //   setUsers((prevUsers) => prevUsers.filter((user) => user.id !== userId));
+  // };
+
+  async function deleteUserContact(phoneNumber: string) {
```

Comment:
declare this service in ```src/service``` folder within relevant file
and follow the similar template as other services declared in '''src/service/userSercvice.ts```